### PR TITLE
Update ec2_eni_facts to boto3

### PR DESCRIPTION
**Pull request by wimnat**
##### SUMMARY
Update to boto3

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ec2_eni_facts

##### ANSIBLE VERSION
2.4

##### ADDITIONAL INFORMATION
Updated to boto3 and added return doc

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/ec2_eni_facts.py [hunk 1]
```diff
@@ -66,122 +256,65 @@
...
-    for interface in all_eni:
-        interface_dict_array.append(get_eni_info(interface))
+    # Turn the boto3 result in to ansible_friendly_snaked_names
+    snaked_network_interfaces = []
+    for network_interface in network_interfaces:
+        # Copy the tags because we don't want to snake these
+        if 'TagSet' in network_interface:
+            tag_set_copy = copy.copy(network_interface['TagSet'])
+        snaked_network_interfaces.append(camel_dict_to_snake_dict(network_interface))
+        snaked_network_interfaces['tags'] = boto3_tag_list_to_ansible_dict(tag_set_copy)
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_eni_facts.py]: Ah, it's nice that the syntax checker caught this bug.

This should be `network_interface['tags'] = boto3_tag_list_to_ansible_dict(tag_set_copy)`

**Review by willthames — CHANGES_REQUESTED** [review]

**Comment by gundalow** [comment]
As agreed in 2017-12-07 Core meeting, we've done bulk pep8 fixes across all modules in `devel`.

This was done in https://github.com/ansible/ansible/pull/33682 and https://github.com/ansible/ansible/pull/33690 which have been merged into `devel`. because of this you will need to rebase your PR. Thanks in advance.

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (100/100 (100%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o6.sgmail.github.com ([192.254.113.101]:51745)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.89)
	(envelope-from <bounces+848413-b37a-greg=biomassiv.es@sgmail.github.com>)
	id 1eNK0r-002laO-46
	for greg@biomassiv.es; Fri, 08 Dec 2017 08:52:29 -0600
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=Go3n57K4oJFCG03qVqjGsLjQ9DA=; b=KtaziKnWpGrdKIh7
	SUl2qjS2EQjcIbGqZyl90Lx/tKkJoE3o9maUkvRty4scz1qDFrHPP1GSosrJm2gX
	Kiwniii8hBjKlZV3nsdVn7cmbK0/j7hbwmAo+8NiCQT4NnqamJIIGUQQ/XkKO+qw
	511x7pUKRSIZHi9rb+6uucNChz0=
Received: by filter0207p1las1.sendgrid.net with SMTP id filter0207p1las1-4729-5A2AA706-15
        2017-12-08 14:51:50.328779596 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd0011p1iad2.sendgrid.net (SG) with ESMTP id EGJubdFIRXKSgoiRZAUSUg
	for <greg@biomassiv.es>; Fri, 08 Dec 2017 14:51:50.218 +0000 (UTC)
Date: Fri, 08 Dec 2017 14:51:50 +0000 (UTC)

**Comment by tremble** [comment]
close_me

https://github.com/ansible/ansible-modules-extras/pull/2425

**Pull request closed without merging**